packages/context/src/binding.ts

```diff
@@ -90,17 +91,55 @@ export enum BindingScope {
   SINGLETON = 'Singleton',
 }
 
+/**
+ * Type of the binding source
+ */
 export enum BindingType {
+  /**
+   * A fixed value
+   */
   CONSTANT = 'Constant',
+  /**
+   * A function to get the value
+   */
   DYNAMIC_VALUE = 'DynamicValue',
+  /**
+   * A class to be instantiated as the value
+   */
   CLASS = 'Class',
+  /**
+   * A provider class with `value()` function to get the value
+   */
   PROVIDER = 'Provider',
 }
 
+// tslint:disable-next-line:no-any
+export type TagMap = MapObject<any>;
```

Comment:
Should `MapObject`'s generic type be of type `string` and not `any`?

Reply:
No. `any` is for tag values, which can be a string, number, boolean, or even objects. The keys of TagMap are tag names (string).

Reply:
Oh, ok. I didn't realize we supported tag values of types other than `string`.